src/tools/execution-results.h

```diff
@@ -73,7 +73,7 @@ struct ExecutionResults {
           // change (after duplicate function elimination or roundtripping)
           // while the function contents are still the same
           for (Literal& val : ret) {
-            if (val.type == Type::funcref) {
+            if (val.type == Type::funcref && !val.isNull()) {
```

Comment:
Another problem I found during debugging, where `(ref.null func)` was replaced with `(ref.func funcref)`, not accounting for `null`.